src/connectors/exchangeRate/index.ts

```diff
@@ -0,0 +1,263 @@
+
+// TYPES
+
+type TokenCurrency = 'LIKE' | 'USDT'
+type FiatCurrency = 'HKD'
+
+type FromCurrency = TokenCurrency | FiatCurrency
+type ToCurrency = 'HKD' | 'TWD' | 'USD'
+
+interface Pair {
+  from: FromCurrency
+  to: ToCurrency
+}
+
+interface Rate extends Pair {
+  rate: number
+  updatedAt: Date
+}
+
+// CONSTANTS
+
+const tokenCurrencies: TokenCurrency[] = ['LIKE', 'USDT']
+const fiatCurrencies: FiatCurrency[] = ['HKD']
+const quoteCurrencies: ToCurrency[] = ['TWD', 'HKD', 'USD']
+
+const TOKEN_TO_COINGECKO_ID = {
+  LIKE: 'likecoin',
+  USDT: 'tether',
+} as const
+
+const COINGECKO_API_URL = 'https://api.coingecko.com/api/v3/simple/price'
+const EXCHANGE_RATES_DATA_API_URL =
+  'https://api.apilayer.com/exchangerates_data/latest'
+
+// TYPE PREDICATES
+
+const isToken = (currency: FromCurrency): currency is TokenCurrency =>
+  tokenCurrencies.includes(currency as any)
+
+const isFiat = (currency: FromCurrency): currency is FiatCurrency =>
+  fiatCurrencies.includes(currency as any)
+
+// MAIN
+
+export class ExchangeRate {
+  cache: CacheService
+  expire: number
+  constructor() {
+    this.cache = new CacheService('exchangeRate')
+    this.expire = CACHE_TTL.STATIC
+  }
+
+  getRates = async (
+    from?: FromCurrency,
+    to?: ToCurrency
+  ): Promise<Rate[] | never> => {
+    const allPairs = [...this.getTokenPairs(), ...this.getFiatPairs()]
+    let pairs = allPairs
+    if (from) {
+      pairs = pairs.filter((p) => p.from === from)
+    }
+    if (to) {
+      pairs = pairs.filter((p) => p.to === to)
+    }
+
+    return Promise.all(pairs.map((p) => this.getRate(p.from, p.to)))
+  }
+
+  updateTokenRates = async () => {
+    await this.updateRatesToCache(await this.fetchTokenRates())
+  }
+
+  updateFiatRates = async () => {
+    await this.updateRatesToCache(await this.fetchFiatRates())
+  }
+
+  getRate = async (
+    from: FromCurrency,
+    to: ToCurrency
+  ): Promise<Rate | never> => {
+    const data = (await this.cache.getObject({
+      keys: this.genCacheKeys({ from, to }),
+      getter: async () => this.fetchAndCacheRate({ from, to }),
+      expire: this.expire,
+    })) as any
+    if (!data) {
+      throw new UnknownError('Unexpected null')
+    }
+    return {
+      from,
+      to,
+      rate: data.rate,
+      updatedAt: new Date(data.updatedAt),
+    }
+  }
+
+  private fetchAndCacheRate = async ({
+    from,
+    to,
+  }: Pair): Promise<Rate | never> => {
+    logger.warn(
+      'exchangeRate requested APIs to get rates instead of from cache'
+    )
+
+    let rate: Rate
+    if (isToken(from)) {
+      const data = await this.requestCoingeckoAPI([from], [to])
+      rate = this.parseCoingeckoData(data, { from, to })
+    } else if (isFiat(from)) {
+      const data = await this.requestExchangeRatesDataAPI(from, [to])
+      rate = this.parseExchangeRateData(data, { from, to })
+    } else {
+      throw new UnknownError('Unknown currency')
+    }
+    await this.updateRatesToCache([rate])
```

Comment:
the `getter` of `getObject` (L88) [already did the `storeObject`](https://github.com/thematters/matters-server/blob/develop/src/connectors/cache/index.ts#L105-L111), so here we can just fetch and return?